src/NuGet.Core/NuGet.CommandLine.XPlat/DeleteCommand.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using NuGet.Commands;
+using NuGet.Configuration;
+using NuGet.Logging;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Core.v3;
+using System.Threading;
+
+namespace NuGet.CommandLine.XPlat
+{
+    class DeleteCommand : Command
+    {
+        public DeleteCommand(CommandLineApplication app, Func<ILogger> getLogger)
+        {
+            app.Command("delete", delete =>
+            {
+                delete.Description = Strings.Delete_Description;
+
+                var source = delete.Option(
+                    "-s|--source <source>",
+                    Strings.Source_Description,
+                    CommandOptionType.SingleValue);
+
+                //TODO: communicate to Nuget team, this should go to a top level argument, and
+                //should not be handled by individual cmdlet
+                var nonInteractive = delete.Option(
+                    "--non-interactive <nonInteractive>",
+                    Strings.NonInteractive_Description,
+                    CommandOptionType.NoValue);
+
+                var argRoot = delete.Argument(
+                    "[root]",
+                    Strings.Delete_PackageIdAndVersion_Description,
+                    multipleValues: true);
+
+                delete.OnExecute(async () =>
+                {
+                    ILogger logger = getLogger();
+                    ISettings setting = Settings.LoadDefaultSettings(Path.GetFullPath("."),
+                                configFileName: null,
+                                machineWideSettings: null);
+                    PackageUpdateResource pushCommandResource = await GetPushCommandResource(source, setting);
+
+                    var packageId = argRoot.Values[0];
+                    var packageVersion = argRoot.Values.Count > 1 ? argRoot.Values[1] : string.Empty;
```

Comment:
If version is left off it just deletes all versions?

All of these parameters need validation. I don't see any checks to validate if the version passed in is a valid NuGetVersion, or that the api key is a key. It looks like down farther is just concats the strings and sends it off to the server or local disk.